app/models/league.rb

```diff
@@ -29,7 +29,11 @@ def add_test_teams
   end
 
   def generate_schedule
-    calculate_round_robin
+    if game_mode == League.game_modes[:round_robin]
+      calculate_round_robin
+    elsif game_mode == League.game_modes[:two_halfs]
+      calculate_double_round_robin
```

Comment:
Are double round Robin and two halfs the same? Then we should probably use only one of these names in the code